scanpy/tools/_tsne.py

```diff
@@ -1,48 +1,59 @@
 from typing import Optional, Union
 
+import numpy as np
+import openTSNE
 from anndata import AnnData
 
-from .._utils import _doc_params, AnyRandom
-from ..tools._utils import _choose_representation, doc_use_rep, doc_n_pcs
-from .._settings import settings
 from .. import logging as logg
+from .._settings import settings
+from .._utils import AnyRandom, NeighborsView
+from ..tools._utils import get_init_pos_from_paga
+
+
+class UniformAffinities(openTSNE.affinity.Affinities):
+    def __init__(self, neighbors, symmetrize=True, verbose=False):
+        self.verbose = verbose
+
+        # Ignore the weights and just assign every neighbor equal weight
+        P = neighbors > 0
+        # Symmetrize the probability matrix
+        if symmetrize:
+            P = (P + P.T) / 2
+        # Convert weights to probabilities
+        P /= np.sum(P)
+
+        self.P = P
```

Comment:
I believe this will error if `symmetrize` is `False`, since you can't do an inplace type conversion. That is, if `symmetrize` is `False`, `P.dtype` is `bool`, so you can't inplace assign floats.

<details>
<summary> Example </summary>

```python
from scipy import sparse
s = sparse.random(100, 100, format="csr", density=0.1)
sb = s > 0
sb /= sb.sum()
```

```pytb
---------------------------------------------------------------------------
UFuncTypeError                            Traceback (most recent call last)
<ipython-input-4-23169c7c63c6> in <module>
----> 1 sb /= sb.sum()

/usr/local/lib/python3.8/site-packages/scipy/sparse/data.py in __itruediv__(self, other)
     61         if isscalarlike(other):
     62             recip = 1.0 / other
---> 63             self.data *= recip
     64             return self
     65         else:

UFuncTypeError: Cannot cast ufunc 'multiply' output from dtype('float64') to dtype('bool') with casting rule 'same_kind'
```

</details>